Core/include/Acts/Utilities/KDRange.hpp

```diff
@@ -0,0 +1,430 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+
+namespace Acts {
+
+/// @brief A one-dimensional range between two points.
+///
+/// This type describes a one-demensional range of values, designed to be used
+/// in the construction of more complex multi-dimensional types. The class
+/// provides functionality for growing and shrinking the range, as well as some
+/// other utilities.
+///
+/// @tparam Type The scalar type of the values contained in this range.
+template <typename Type>
+class OneDRange {
```

Comment:
I see, this could actually be useful also in the Geometry parsing ... 

... we called the `KDBinnedArray`  actually `BinnedArrayXD` - so for consistency, one could do the same here and call this `RangeXD`, while the actual classes could be called `Range1D`, `Range2D` ... etc.